grouper/graph.py

```diff
@@ -141,25 +142,34 @@ def _get_user_metadata(session):
         '''
         Returns a dict of username: { dict of metadata }.
         '''
-        users = session.query(User)
 
-        public_keys = {}
-        for key in session.query(PublicKey):
-            if key.user_id not in public_keys:
-                public_keys[key.user_id] = []
-            public_keys[key.user_id].append(key)
+        def user_indexify(data):
+            ret = {}
```

Comment:
`ret = defaultdict(list)` will save you a few lines. you could also replace the entire block with a `reduce` which is arguably less readable.